mysql-test/mysql-test-run.pl

```diff
@@ -1719,17 +1720,28 @@ sub command_line_setup {
     $opt_valgrind= 1;
     $opt_valgrind_mysqld= 1;
 
+    my @valgrind_args_orig= @valgrind_args;
+
+    unshift(@valgrind_args, "--tool=callgrind");
     # Set special valgrind options unless options passed on command line
     push(@valgrind_args, "--trace-children=yes")
-      unless @valgrind_args;
+      unless @valgrind_args_orig;
   }
 
-  if ( $opt_valgrind )
+  # default to --tool=memcheck
+  if ($opt_valgrind && ! grep(/^--tool=/i, @valgrind_args))
   {
     # Set valgrind_options to default unless already defined
-    push(@valgrind_args, @default_valgrind_args)
+    push(@valgrind_args, @default_memcheck_valgrind_args)
       unless @valgrind_args;
+    push(@valgrind_args, "--suppressions=${glob_mysql_test_dir}/valgrind.supp")
+      if -f "$glob_mysql_test_dir/valgrind.supp";
```

Comment:
I agree moving to global for all tools seems worth while. Each suppression seems to be tagged with the tool so there shouldn't be regressions (assuming correct valgrind implementation).